File: CommonLibrary/src/main/java/com/common/util/SPUtils.java
package com.common.util;

import android.content.Context;
import android.content.SharedPreferences;

/**
 * by y on 2017/5/18
 */

public class SPUtils {
    private SPUtils() {
    }

    private static final String SHAREDPREFERENCES_NAME = "codeKK";
    private static SharedPreferences sharedPreferences;

    public static final String IS_OP_TAG = "op_tag";
    public static final String IS_OP_URL_WEB = "op_tvWeb";
    public static final String IS_OPA_TAG = "opa_tag";
    public static final String IS_OPA_URL_WEB = "opa_web";
    public static final String IS_BLOG_TAG = "blog_tag";

    private static void initSharePreferences(Context context) {
        sharedPreferences = context.getSharedPreferences(SHAREDPREFERENCES_NAME, Context.MODE_PRIVATE);
    }

    public static void init(Context context) {
        initSharePreferences(context.getApplicationContext());
    }

    public static boolean isNull() {
        return sharedPreferences == null;
    }

    public static boolean getBoolean(String key, boolean defaultValue) {
        if (isNull()) {
            return defaultValue;
        }
        return sharedPreferences.getBoolean(key, defaultValue);
    }

    public static void setBoolean(String key, boolean value) {
        if (isNull()) {
            return;
        }
        sharedPreferences.edit().putBoolean(key, value).apply();
    }

    public static String getString(String key, String defaultValue) {
        if (isNull()) {
            return defaultValue;
        }
        return sharedPreferences.getString(key, defaultValue);
    }

    public static void setString(String key, String value) {
        if (isNull()) {
            return;
        }
        sharedPreferences.edit().putString(key, value).apply();
    }

    public static void clearAll() {
        if (isNull()) {
            return;
        }
        sharedPreferences.edit().clear().apply();
    }

}

File: commonLibrary/src/main/java/com/common/util/SPUtils.kt
package com.common.util

import android.content.Context
import android.content.SharedPreferences

/**
 * by y on 2017/5/18
 */

object SPUtils {

    private const val SHAREDPREFERENCES_NAME = "codeKK"
    private lateinit var sharedPreferences: SharedPreferences

    const val IS_OP_TAG = "op_tag"
    const val IS_OP_URL_WEB = "op_tvWeb"
    const val IS_OPA_TAG = "opa_tag"
    const val IS_OPA_URL_WEB = "opa_web"
    const val IS_BLOG_TAG = "blog_tag"

    private fun initSharePreferences(context: Context) {
        sharedPreferences = context.getSharedPreferences(SHAREDPREFERENCES_NAME, Context.MODE_PRIVATE)
    }

    fun init(context: Context) {
        initSharePreferences(context.applicationContext)
    }

    fun getBoolean(key: String, defaultValue: Boolean): Boolean {
        return sharedPreferences.getBoolean(key, defaultValue)
    }

    fun setBoolean(key: String, value: Boolean) {
        sharedPreferences.edit().putBoolean(key, value).apply()
    }

    fun getString(key: String, defaultValue: String): String? {
        return sharedPreferences.getString(key, defaultValue)
    }

    fun setString(key: String, value: String) {
        sharedPreferences.edit().putString(key, value).apply()
    }

    fun clearAll() {
        sharedPreferences.edit().clear().apply()
    }

}
